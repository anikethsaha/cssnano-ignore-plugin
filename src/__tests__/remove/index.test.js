const fs = require('fs');
const path = require('path');
const postcss = require('postcss');
const plugin = require('../../remove');

function run(fixturename) {
  it(
    'should match the snapshot for ' + fixturename.replace('.css', ''),
    (done) => {
      const input = fs.readFileSync(
        path.join(__dirname, `./fixtures/${fixturename}`),
        'utf8'
      );
      let result = postcss([plugin.default()]).process(input, {
        from: undefined,
      });
      expect(result.css).toMatchSnapshot();
      done();
    }
  );
}

fs.readdirSync(path.join(__dirname, './fixtures')).forEach((fixtureName) => {
  run(fixtureName);
});
