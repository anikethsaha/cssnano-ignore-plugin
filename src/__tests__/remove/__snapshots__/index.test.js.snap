// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for atRule 1`] = `
"@media screen and (min-width: 480px) {
  body {
    background-color: lightgreen;
  }
  a {
  }
}
"
`;

exports[`should match the snapshot for basic 1`] = `
"#main,
ul li {
  margin: auto;
}
"
`;

exports[`should match the snapshot for rules 1`] = `
"a {
  color: red;
}

p {
  padding: 10px;
}

#main {
  border: 1px solid black;
}
"
`;

exports[`should match the snapshot for rulesInAtRules 1`] = `
"@media screen and (min-width: 900px) {
}

@media screen and (min-width: 900px) {
  body {
    padding: 1rem 3rem;
  }
}
@document url(\\"https://www.example.com/\\") {
  h1 {
    color: green;
  }
}

@media screen and (min-width: 900px) {
  html {
    padding: 1rem 3rem;
  }
}
"
`;

exports[`should match the snapshot for wrongComment 1`] = `
"#main,
ul li {
  /* postcss-ignore-line-wrong-comment */
  border: 1px solid black;
  margin: auto;
}
"
`;
